File: VB/Data/ReportCatalogDbContext.vb
Imports System.ComponentModel.DataAnnotations
Imports System.Data.Entity

Namespace ShowReportPreview_MVVM.Data
	Public Class ReportCatalogItem
		<Key>
		Public Property ID() As String
		Public Property ReportName() As String
		Public Property Layout() As Byte()
	End Class

	Public Class ReportCatalogDbContext
		Inherits DbContext

		Shared Sub New()
			Database.SetInitializer(New NullDatabaseInitializer(Of ReportCatalogDbContext)())
		End Sub

		Public Sub New()
			MyBase.New("name=ReportCatalogDbContext")
		End Sub

		Public Property ReportItems() As DbSet(Of ReportCatalogItem)
	End Class
End Namespace

File: CS/Data/ReportCatalogDbContext.cs
using System.ComponentModel.DataAnnotations;
using System.Data.Entity;

namespace ShowReportPreview_MVVM.Data {
    public class ReportCatalogItem {
        [Key]
        public string ID { get; set; }
        public string ReportName { get; set; }
        public byte[] Layout { get; set; }
    }

    public class ReportCatalogDbContext : DbContext {
        static ReportCatalogDbContext() {
            Database.SetInitializer(new NullDatabaseInitializer<ReportCatalogDbContext>());
        }

        public ReportCatalogDbContext() : base("name=ReportCatalogDbContext") { }

        public DbSet<ReportCatalogItem> ReportItems { get; set; }
    }
}
